Text/Distinct/Thesaurus.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.  
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.  
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.Distinct
+{
+    public class Thesaurus
+    {
+        public Thesaurus(string executingDirectoryPath)
+            : this(JsonConvert.DeserializeObject<IEnumerable<IEnumerable<string>>>(
+                File.ReadAllText($"{executingDirectoryPath}\\thesaurus.json")))
+        { }
+
+        public Thesaurus(IEnumerable<IEnumerable<string>> dataset)
+        {
+            Synonyms = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
+            foreach (IEnumerable<string> lemma in dataset)
+            {
+                string canonicalForm = lemma.First();
+                foreach (string form in lemma)
+                {
+                    Synonyms.Add(Normalize(form), canonicalForm);
```

Comment:
>Add [](start = 29, length = 3)

this throws if the element is already present - which might happen since you're normalizing the forms?

may want to check if it already exists, and if so is the value the same as this new canonicalForm - and only throw if they mismatch (ambiguity of intent)